### ISSUE TITLE
[BUG] New manga view Share URL wrong

### ISSUE THREAD

**Issue by henderea** [Paperback-iOS/app#551]
**Describe the bug**
The share button in the new view seems to be broken.  The URLs it gives me for MangaDex are not valid.  The ID getting passed into the `getMangaShareUrl` method doesn’t seem to match the ID used for looking up the manga data in the API.

Example:
MangaDex search result: https://mangadex.org/title/37536724-111d-4a99-b13c-76ec22600789/kamigami-no-kago-de-seisan-kakumei
Share URL: https://mangadex.org/title/1253F2DE-128B-457B-97AA-F34BD595B234

I have no idea where that ID for the share URL is coming from, but those are the same manga.  If I look at the logs for when I refresh the chapters, I see:
```
[INFO] [2022-04-23T19:40:33.706Z] [AsyncOperation:start():51] [AsyncOperation] RequestOperation - https://api.mangadex.org/manga/37536724-111d-4a99-b13c-76ec22600789/feed?limit=500&includes[]=scanlation_group&translatedLanguage[]=en&order[volume]=desc&order[chapter]=desc&order[publishAt]=desc&contentRating[]=erotica&contentRating[]=suggestive&contentRating[]=safe&contentRating[]=pornographic&includeFutureUpdates=0
```

Which is the correct ID.  Yet the share URL uses some other ID that doesn’t seem to match a manga.

This issue seems to be specific to the new view. The old view seems to still be opening up the page fine.

**To Reproduce**
Try to get the share URL of a search result from MangaDex (when using the new manga view), for example, `Kamigami no Kago de Seisan Kakumei` (the manga from above).  It won’t make a valid URL because the ID doesn’t exist.  Note that this issue is only in the new manga view.

**Expected behaviour**
The share URL should use the ID of the manga instead of some random ID from who knows where. It works fine in the old view, so this seems to be a bug in the new view.

**App Version**
v0.7-r26 (also in r25; r24 didn’t have a working share button to test with)

**Screenshots**

**Comment by TheNetsky** [comment]
This is not really a app issue.

**Comment by henderea** [comment]
As I said, it works fine in the old manga view. With the exact same manga and source. It's only broken in the new manga view. So it's definitely an app bug.

**Comment by henderea** [comment]
I had looked at the code for the source, and it uses the `mangaId` parameter as-is for both the share URL and the data fetches. But somehow, the new view gives a different id for getting the share url than what it uses for refreshing the chapter list.

**Comment by FaizanDurrani** [comment]
This is indeed an app issue and will be fixed in the next update.

**Comment by FaizanDurrani** [comment]
Fixed in 0.7-r31
